pkg/controller/machine/controller.go

```diff
@@ -58,6 +58,10 @@ func newReconciler(mgr manager.Manager, actuator Actuator) reconcile.Reconciler
 		actuator: actuator,
 	}
 
+	mgr.GetFieldIndexer().IndexField(&clusterv1.Cluster{}, "metadata.name", func(obj runtime.Object) []string {
```

Comment:
Does it matter that IndexerFunc is meant to be unique across namespaces? https://github.com/kubernetes-sigs/cluster-api/blob/0617f27a880dc94236351ad9f53ea70ea21ca308/vendor/sigs.k8s.io/controller-runtime/pkg/client/interfaces.go#L91

Reply:
I believe that's taken care of in this wrapper https://github.com/kubernetes-sigs/cluster-api/blob/0617f27a880dc94236351ad9f53ea70ea21ca308/vendor/sigs.k8s.io/controller-runtime/pkg/cache/informer_cache.go#L133-L178

Reply:
@vincepri From the looks of it, that field indexes both the namespaced and non-namespaced keys for each value, so wouldn't there be a collision for the non-namespaced value of two clusters with the same name in different namespaces?